panther_analysis_tool/schemas.py

```diff
@@ -65,12 +65,12 @@
         },
         Optional('Tests'): [{
             'Name': str,
-            'ResourceType': str,
             'ExpectedResult': bool,
             'Resource': object,
         }],
     },
-    ignore_extra_keys=False)
+    # For backwards compatibility with older schemas having more fields
+    ignore_extra_keys=True)
```

Comment:
My concern with `ignore_extra_keys=True` is that then any keys could be added, not just the test fields right? Perhaps it would be better to just leave the `ResourceType` and `LogType` fields in the schemas but change them to optional. Then we could still catch extraneous fields that are not just there for backwards compatibility, and also allow the backwards compatible fields to stay in.

Reply:
Yes, all extra fields would be ignored. I couldn't think of any downside for this generally. Maybe if the `analysis-api` is designed the same way and crashes if it sees extra fields? Will check. Could you think of any other issue with ignoring all extra fields?

Thanks for reviewing!

Reply:
The main reason is to protect users against mistakes, if they put in a typo like `Descrpition`, no errors would be raised if we accept additional fields because `Description` is optional in the first place and `Descrpition` would just be ignored as an extraneous field. Or alternatively if they put in `Run book` instead of `Runbook`, or they call `Test` instead of `Tests`, those types of mistakes. Perhaps that is being too coddling to users, but I think it is helpful.

I don't think it would cause any issues on the backend in the `analysis-api` though, in fact in the past we've added fields to the policies/rules & `panther_analysis_tool` first before adding them to the `analysis-api` and it didn't cause any issues.

Reply:
Makes sense! Reverted to fail on extra keys and made the specific fields optional.